Mathlib/Topology/Algebra/Order/UpperLower.lean

```diff
@@ -59,7 +60,38 @@ instance (priority := 100) OrderedCommGroup.to_hasUpperLowerClosure [OrderedComm
 #align ordered_comm_group.to_has_upper_lower_closure OrderedCommGroup.to_hasUpperLowerClosure
 #align ordered_add_comm_group.to_has_upper_lower_closure OrderedAddCommGroup.to_hasUpperLowerClosure
 
-variable [Preorder α] [HasUpperLowerClosure α] {s : Set α}
+variable [Preorder α]
+
+section OrderClosedTopology
+variable [OrderClosedTopology α] {s : Set α}
+
+@[simp] lemma upperBounds_closure (s : Set α) : upperBounds (closure s : Set α) = upperBounds s :=
+  ext fun a ↦ by simp_rw [mem_upperBounds_iff_subset_Iic, isClosed_Iic.closure_subset_iff]
+#align upper_bounds_closure upperBounds_closure
```

Comment:
Please move this to the recently created `Topology/Order/OrderClosed` and assume `ClosedIicTopology` instead of `OrderClosedTopology`.